pkg/source/source.go

```diff
@@ -75,6 +75,7 @@ var defaultSources Sources = []Source{
 	LocalAPI,
 	Generated,
 	Restored,
+	Directory,
```

Comment:
Where are the users for `Directory`? There is no context and no user right now, so I don't see how anyone currently or in future can reason about whether this is the right priority level for resources sourced from a directory.

This commit seems like it should be a dedicated PR that also introduces the functionality to import resources from a directory, not just a symbol with the test.

Reply:
> Where are the users for Directory? There is no context and no user right now,
> 
> This commit seems like it should be a dedicated PR that also introduces the functionality to import resources from a directory, not just a symbol with the test.

My understanding was that such functionality was added in https://github.com/cilium/cilium/pull/32599, but the authors forgot to add the new `Source` to `AllowOverwrite()` or tests, so I wanted to rectify things here.

> so I don't see how anyone currently or in future can reason about whether this is the right priority level for resources sourced from a directory.

Since `Directory` has not been mentioned explicitly in `AllowOverwrite`, it can be overwritten by any Source, which is why I placed it just above `Unspec`.

Reply:
I've isolated these changes to https://github.com/cilium/cilium/pull/36095, we can continue the discussion there.